### PR TITLE
Fix previewing a file w/ no functions

### PR DESCRIPTION
**Pull request by jasonLaster**
### Summary of Changes

- Previously we assumed files would always have one function if we wanted to preview them.
- now we check for atleast one identifier, which is more general.

### CODE REVIEW

**Comment by nchevobbe** [comment]
could we add a test to prevent regressing this ?

File: src/test/mochitest/browser.ini [hunk 1]
```diff
@@ -138,6 +140,8 @@ skip-if = os == "win"
 [browser_dbg-pretty-print-paused.js]
 [browser_dbg-preview.js]
 skip-if = os == "win"
+[browser_dbg-preview-module.js]
+skip-if = os == "win"
```
Comment (nchevobbe) [inline src/test/mochitest/browser.ini]: why do we skip on windows ?

File: src/test/mochitest/browser_dbg-preview-module.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+// Test hovering in a script that is paused on load
+// and doesn't have functions.
+add_task(async function() {
+  const dbg = await initDebugger("doc-scripts.html");
+  const { selectors: { getSelectedSource }, getState } = dbg;
+
+  navigate(dbg, "doc-on-load.html")
+
+  await waitForPaused(dbg);
```
Comment (nchevobbe) [inline src/test/mochitest/browser_dbg-preview-module.js]: could you add a comment to say that the test page has a debugger statement that will pause the debugger, hence the waitForPaused here ?

File: src/test/mochitest/browser_dbg-preview-module.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  const popupPreviewed = waitForDispatch(dbg, "SET_PREVIEW");
+  hoverAtPos(dbg, { line: 1, ch: 6 });
+  await popupPreviewed;
+  await assertPopup(dbg, { field: "foo", value: "1", expression: "obj" });
+  await assertPopup(dbg, { field: "bar", value: "2", expression: "obj" });
+
+  // hover over an empty position
```
Comment (nchevobbe) [inline src/test/mochitest/browser_dbg-preview-module.js]: I suspect this is done to hide the tooltip ? Would you mind adding a comment or an info call to explain that ?

File: src/test/mochitest/head.js [hunk 4]
```diff
@@ -1084,6 +1084,47 @@ function getCM(dbg) {
   return el.CodeMirror;
 }
 
+
+function getCoordsFromPosition(cm, { line, ch }) {
+  return cm.charCoords({ line: ~~line, ch: ~~ch });
```
Comment (nchevobbe) [inline src/test/mochitest/head.js]: I had to lookup this :) So this is done to have a Math.floor, do we ever trigger this with floating numbers ?

File: src/test/mochitest/head.js [hunk 5]
```diff
@@ -1084,6 +1084,47 @@ function getCM(dbg) {
...
+  const previewEl = await waitForElement(dbg, "tooltip");
+  is(previewEl.innerText, result, "Preview text shown to user");
+
+  const preview = dbg.selectors.getPreview(dbg.getState());
+  is(`${preview.result}`, result, "Preview.result");
+  is(preview.updating, false, "Preview.updating");
+  is(preview.expression, expression, "Preview.expression");
+}
+
+async function assertPopup(dbg, { field, value, expression }) {
```
Comment (nchevobbe) [inline src/test/mochitest/head.js]: nit: Could this be assertPreviewPopup ? We do have other things that might be seen as "Popup" (like, the Help modal, even if we call it a modal :D )

File: src/test/mochitest/browser.ini (hunk 1 above)
Comment (jasonLaster) [reply src/test/mochitest/browser.ini]: windows is weird w/r/t to the mouse events

File: src/test/mochitest/head.js (hunk 4 above)
Comment (jasonLaster) [reply src/test/mochitest/head.js]: I believe this helped before and it is copy pasta from elsewhere

**Pull request merged**